src/sql/workbench/browser/editor/tableDesigner/tableDesignerInput.ts

```diff
@@ -56,7 +56,11 @@ export class TableDesignerInput extends EditorInput {
 				this._onDidChangeDirty.fire();
 			}
 		}));
-		this._tableIcon = tableInfo.tableIcon.toString() === '' ? TableIcon.Basic : tableInfo.tableIcon as TableIcon;
```

Comment:
this should work: `this._tableIcon = tableInfo.tableIcon ? tableInfo.tableIcon as TableIcon : TableIcon.Basic ;`

Reply:
No it will fail for empty string (`tableInfo.tableIcon === ''`)

Reply:
are you sure? if tableIcon is empty string, it will be evaluated to false.

Reply:
Yeah I'm confused why that wouldn't work. And you should need to cast, they're both `TableIcon`.

`this._tableIcon = tableInfo.tableIcon || TableIcon.Basic;`

Reply:
The empty string is not mapped to anything. The issue is because the type is TableIcon and not string.